csm_web/frontend/src/components/App.js

```diff
@@ -40,13 +40,32 @@ export default class App extends React.Component {
   }
 }
 
+const applyDarkMode = isDarkMode => {
+  window.localStorage.setItem("darkMode", isDarkMode);
+  document.getElementById("app").className = isDarkMode ? "dark-mode" : "";
+  document.body.style.backgroundColor = isDarkMode ? "#1c1d1e" : "white";
+};
+
 function Header() {
+  const [isDarkMode, setDarkMode] = React.useState(window.localStorage.getItem("darkMode") == "true");
```

Comment:
same as above. dont do window.localstorage stuff

Reply:
This is required to persist the theme across reloads.